lib/jxl/encode.cc

```diff
@@ -562,6 +563,15 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderSetExtraChannelInfo(
   if (index >= enc->metadata.m.num_extra_channels) {
     return JXL_API_ERROR("Invalid value for the index of extra channel");
   }
+  if (!info->exponent_bits_per_sample) {
+    if (!(info->bits_per_sample > 0 && info->bits_per_sample <= 24)) {
+      return JXL_API_ERROR("Invalid value for bits_per_sample");
+    }
+  } else if ((info->exponent_bits_per_sample > 8) ||
+             (info->bits_per_sample > 24 + info->exponent_bits_per_sample)) {
+    return JXL_API_ERROR("Invalid float description");
```

Comment:
Better like this?

Reply:
Maybe it's best to have an auxiliary function `CheckValidBitdepth(bits_per_sample, exponent_bits_per_sample)` that gets used both here (for extra channels) and when checking those fields for the color channels. For now we can keep the overly strict behavior, and then do a follow-up MR that changes the behavior to allow anything the bitstream allows (currently the code is confusing what the pixelformat allows and what the bitstream allows).

Reply:
sounds like a plan!

Reply:
When checking those fields for the color channels, we also call `SetFooSamples`, which we do manually for the alpha, but an additional check won't hurt, I guess.

Reply:
follow up here: #977